ztools/ztls/handshake_messages.go

```diff
@@ -652,9 +681,24 @@ func (m *serverHelloMsg) marshal() []byte {
 		z[1] = byte(extensionExtendedMasterSecret & 0xff)
 		z = z[4:]
 	}
+	if sctLen > 0 {
+		z[0] = byte(extensionSCT >> 8)
+		z[1] = byte(extensionSCT)
+		l := sctLen + 2
```

Comment:
Are you sure this isn't plus 4? The length of `sctLen` includes the length of the length bytes?